### DISCUSSION TITLE
NEXTJS API route gives 404 on POST

### DISCUSSION THREAD

**Discussion by juhamm** [vercel/next.js#41405] [Help]
I'm having an issue with API routes and POST. This route is located in /pages/api/uploadProductimages.js For some reason POST gives me 404, however if I try GET it works fine??

here is /pages/api/uploadProductimages.js

```
export default function handler(req, res) {
    if (req.method === 'POST') {
       res.status(200).json({name: "POST"})
    } else {
        res.status(200).json({name: "GET"})
    }
  }
```
  
This works fine:

```
const res = await axios.get(`/api/uploadProductimages`);
console.log(res);
```

But this gives me 404 :(( http://localhost:3001/api/uploadProductimages 404 (Not Found)
```
const res = await axios.post(`/api/uploadProductimages`, formData,{
              headers: {
                'Content-Type': 'multipart/form-data'
              },
            });
console.log(res); // this gives 404 
```
POST http://localhost:3001/api/uploadProductimages 404 (Not Found)

I've nextjs 10.2.3, react 16.14.0 + I've custom server (server.js) as well where is other API:s like this

`server.post('/api/v1/product', (req,res) => { ... `

And it works fine

**Comment by icyJoseph** [comment]
Perhaps you have some kind of custom-middleware that checks if a request has POST method, then it should also have content-type set to application/json or something like that?

**Reply by juhamm** [reply]
I checked, if there is some Middware, but not :( And I tried to change 'Content-Type': 'multipart/form-data' to  'Content-Type': 'application/json' but I still get 404 with Post:(

**Reply by icyJoseph** [reply]
perhaps you client is sending "post" lowercased? Sort of an edge case, but would be good to log `req.method` on your handler, perhaps it is not even reachable for some reason?

**Reply by juhamm** [reply]
It Is Capitalised POST, and I tried API ROUTE like this: 
```
export default (req, res) => {
    const { method } = req
    switch (method) {
      case 'GET':
        res.status(200).json({name: "GET"})
        break
      case 'POST':
        res.status(200).json({name: "POST"})
        break
      default:
        res.setHeader('Allow', ['GET', 'POST'])
        res.status(405).end(`Method ${method} Not Allowed`)
  }
}
```
But still I get 404.
I just wonder if problem is the next version? I've nextjs 10.2.3 and react react 16.14.0

**Reply by icyJoseph** [reply]
You still get 404? And not 405? That's really strange. AFAIK, there's no bug like that on Next 10.x.x, and it is not so trivial to go through the issues history. 

You did mention you have a custom server right? What if you removed it for sake of testing with the default server? 

The React version, at least at this point and unless this is a mad bug, shouldn't matter at all. 

This is purely API routes handling. How are you using this API? From the client right? Are you doing any redirects? Rewrites? 

Last but not least, got a repository I could take a look at?
